File: database/database-annotation-processor/src/main/java/io/koraframework/database/annotation/processor/jdbc/extension/JdbcTypesExtensionFactory.java
package io.koraframework.database.annotation.processor.jdbc.extension;

import io.koraframework.database.annotation.processor.jdbc.JdbcTypes;
import io.koraframework.kora.app.annotation.processor.extension.ExtensionFactory;
import io.koraframework.kora.app.annotation.processor.extension.KoraExtension;

import javax.annotation.processing.ProcessingEnvironment;
import java.util.Optional;

public class JdbcTypesExtensionFactory implements ExtensionFactory {

    @Override
    public Optional<KoraExtension> create(ProcessingEnvironment processingEnvironment) {
        var type = processingEnvironment.getElementUtils().getTypeElement(JdbcTypes.JDBC_ENTITY.canonicalName());
        if (type == null) {
            return Optional.empty();
        }

        return Optional.of(new JdbcTypesExtension(processingEnvironment));
    }
}

File: database/database-symbol-processor/src/main/kotlin/io/koraframework/database/symbol/processor/jdbc/extension/JdbcTypesExtensionFactory.kt
package io.koraframework.database.symbol.processor.jdbc.extension

import com.google.devtools.ksp.processing.CodeGenerator
import com.google.devtools.ksp.processing.KSPLogger
import com.google.devtools.ksp.processing.Resolver
import io.koraframework.kora.app.ksp.extension.ExtensionFactory

class JdbcTypesExtensionFactory : ExtensionFactory {
    override fun create(resolver: Resolver, kspLogger: KSPLogger, codeGenerator: CodeGenerator) = JdbcTypesExtension()
}
